feature-libs/user/account/components/login-form/login-form.module.ts

```diff
@@ -36,8 +39,8 @@ import { LoginFormComponent } from './login-form.component';
           providers: [
             {
               provide: LoginFormComponentService,
-              useClass: LoginFormComponentService,
-              deps: [AuthService, GlobalMessageService, WindowRef],
+              useClass: environment.cdc? CdcLoginComponentService: LoginFormComponentService,
```

Comment:
I think this should be handled somehow inside CDC lib. To not touch any other file outside CDC integration lib.

Reply:
@dydome,

I tried overriding providers within the cdc lib, but it does not load the intended service.
Any thoughts on how to implement this from cdc lib?

Reply:
Do I understand correctly your problem? :
We want to replace the default implementation of the component service, using the cdc-powered version of this service? - and only when the CDC feature library is installed?

If yes, I would need to think about the possible solutions to this problem... And take into account the lazy-loaded nature of the @spartacus/user/account components.


--
Side note: Using `environment` should be never used in the liibraries code (it can be used only in the /storefrontapp just for the purposes of running our example shellapp with different configuration).
It's becasue that after we build the library, in the output code there is no more any condition on the `environment` conditions. When we build our libs, the environment is resolved during the build and the conditionaly resolved code is "fixed" into the build result in `/dist` folder and then shipped to NPM . So when customers  download our lib to their node_modules, it would have the fixed version of our code, without any condition on the `environment`. And I believe that's not what we wanted to achieve

Reply:
Hi @Platonn,

Yes that is exactly what we need to achieve.
If the CDC feature is enabled, it should invoke our service that calls CDC SDK & APIs for login and registration.


Oh, thanks for letting me know about the caveat of  `environment`.

Initially, I tried to capture events, but they do not pass on sensitive information like passwords.
Another alternative is to create a new endpoint and mimic login / registration screens, but we already have an endpoint `/cdc/login` for CDC custom UI based login / registration.

Any suggestions here is appreciated!

Thanks